promql/testdata/matrix.test

```diff
@@ -0,0 +1,10 @@
+# Matrix tests.
```

Comment:
These are selectors, so can go in this file. Not sure why this is called matrix.

I believe these tests relate to a poor hashing algorithm we used many years back.

Reply:
Moved.